### PR TITLE
Synced with template and updated box version 0.6.0

### LINKED ISSUE #34: Sync with template and update to use 0.6.0 box

### PR DESCRIPTION
**Pull request by claesgill**
Closes #34 

- Synced with template
- Updated box version 0.4.0 -> 0.6.0

### CODE REVIEW

File: dev/vagrant/conf/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ There are two layers of configuration built into the box.
 You may add a `pre_ansible.sh` script file to this directory to run any alterations **before** ansible bootstrap procedure will run.
 
 This might come handy if you need to change or replace that bootstrap process. For example you replacing the entire `/etc/ansible` directory.
-For most cases, you are probably looking to add configuration in [pre/poststart bootstrap](pre_bootstrap/README.md)
+For most cases, you are probably looking to add configuration in [pre/poststart bootstrap](../pre_bootstrap/README.md)
```
Comment (zhenik) [inline dev/vagrant/conf/README.md]:
```suggestion
For most cases, you are probably looking to add configuration in [pre/pre_bootstrap](../pre_bootstrap/README.md)
```

File: template_example/dev/vagrant/bootstrap/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Customize the startup procedure of the Hashistack with pre- and post start ansible scripts
+
+You may put any number of script files in this directory for running ansible commands prior to bootstrapping the hashistack.
+The bootstrap procedure is included/hardcoded in your box.
```
Comment (zhenik) [inline template_example/dev/vagrant/bootstrap/README.md]:
```suggestion
The bootstrap procedure is included/hardcoded in vagrant-hashistack box.
```

**Review by zhenik — CHANGES_REQUESTED** [review]

File: dev/vagrant/conf/README.md (hunk 1 above)
Comment (claesgill) [reply dev/vagrant/conf/README.md]: These should probably  be fixes in the template aswell

File: template_example/dev/vagrant/bootstrap/README.md (hunk 2 above)
Comment (claesgill) [reply template_example/dev/vagrant/bootstrap/README.md]: These should probably  be fixes in the template aswell

**Comment by claesgill** [comment]
Regarding the execution sequence: 

Should [00-create_dir_for_persisting_data.yml](dev/vagrant/bootstrap/nomad/pre/00-create_dir_for_persisting_data.yml) and [022-update-nomad-write-default-policy.yml](dev/vagrant/bootstrap/nomad/post/022-update-nomad-write-default-policy.yml) be named 030 and 031 instead of 00 and 022?

**Comment by zhenik** [comment]
> Regarding the execution sequence:
> 
> Should [00-create_dir_for_persisting_data.yml](dev/vagrant/bootstrap/nomad/pre/00-create_dir_for_persisting_data.yml) and [022-update-nomad-write-default-policy.yml](dev/vagrant/bootstrap/nomad/post/022-update-nomad-write-default-policy.yml) be named 030 and 031 instead of 00 and 022?

these numbers just to define seq. And try to have some room, you might need in the future. So I recommend `030` and `040`

File: dev/ansible/01_generate_secrets_vault.yml [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+
 - name: Generate Secrets with Vault
   shell: |
       vault kv put secret/postgres \
       username=$(vault write -field='value' gen/password length=12 symbols=0) \
       password=$(vault write -field='value' gen/password length=12 symbols=0)
   run_once: true
   environment:
-      VAULT_ADDR: "http://127.0.0.1:8200"
-      VAULT_TOKEN: $vault_master_token
```
Comment (zhenik) [inline dev/ansible/01_generate_secrets_vault.yml]: you might need to make a lookup, here is example https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/ansible/tests/enterprise/vault_namespaces_test.yml#L16

**Review by zhenik — CHANGES_REQUESTED** [review]

File: dev/ansible/01_generate_secrets_vault.yml (hunk 3 above)
Comment (claesgill) [reply dev/ansible/01_generate_secrets_vault.yml]: This is outdated - created a new task instead that exports `VAULT_TOKEN`. But should probably use lookup anyways :smiley:

**Review by zhenik — DISMISSED** [review]
Tough one. Good job

**Review by zhenik — APPROVED** [review]

**Pull request merged**